plugins/woocommerce/templates/global/quantity-input.php

```diff
@@ -23,29 +23,43 @@
 		<input type="hidden" id="<?php echo esc_attr( $input_id ); ?>" class="qty" name="<?php echo esc_attr( $input_name ); ?>" value="<?php echo esc_attr( $min_value ); ?>" />
 	</div>
 	<?php
-} else {
-	/* translators: %s: Quantity. */
-	$label = ! empty( $args['product_name'] ) ? sprintf( esc_html__( '%s quantity', 'woocommerce' ), wp_strip_all_tags( $args['product_name'] ) ) : esc_html__( 'Quantity', 'woocommerce' );
+}
```

Comment:
It seems a little odd that we are still outputting a hidden input (lines 20-26). Additionally, this leads to us having two inputs with the same ID on the same page (something to avoid where possible).

Reply:
Might help: https://github.com/woocommerce/woocommerce/commit/846e1ca25c078f9803f955e14617f3646d7777f3

If I understand correctly, the goal of this PR is essentially to revert this old commit ^. I agree that there's no need to preserve the hidden input -- but it might make sense to think about Mike's motivation behind doing that old commit.

Reply:
@barryhughes @manospsyx We have hidden input because when minimum and maximum quantity are same, we do have a dud quantity field input box, which we do by setting the [disabled attribute to true](https://github.com/woocommerce/woocommerce/pull/34282/files#diff-5fe95244e467e57c0cf6ade504e454fb98ae2f474351186612da8e096eed4195R42) on it. If a field has `disabled` attribute, the browser will not send that field's value to the backend on form submission. In that case, we rely on the hidden field to ensure that the quantity field value is sent to the backend as expected.

Reply:
Understood, thanks! 

Even though the name:value of a disabled input will not be submitted, this still leads us to have a duplicate ID in the DOM.  Not a huge problem, but it's still something we should address if there is a reasonably straightforward way of doing so. What do you think of using the `readonly` property (instead of `disabled`)? This means we can have a single input, the value will be submitted, and it will not normally be possible for end-users to modify it.

Reply:
:writing: `wp_readonly()` is a helper function we could take advantage of for this.

Reply:
@barryhughes Absolutely, that does make sense. Just yesterday, before going to bed, I realised I could've tried `readonly` there, huh 🤔.

Reply:
Great—I think that will be a nice clean solution (and will reduce the total amount of code)—and we can then move this toward completion.